File: Server/src/Application/BorrowRequests/Queries/GetBorrowRequests/BorrowRequestDto.cs
using Domain.BorrowRequests.Enums;

namespace Application.BorrowRequests.Queries.GetBorrowRequests;


public sealed record BorrowRequestDto(
    Guid Id,
    UserSummaryDto Borrower,
    ItemSpecificationDto ItemNeeded,
    TimeSlotDto NeededDates,
    BorrowRequestStatus BorrowRequestStatus,
    DateTimeOffset CreatedAt,
    int offerCount,
    OfferSummaryDto? OfferSummaryDto,
    BorrowRequestActionsDto BorrowRequestActionsDto
    );

public sealed record BorrowRequestActionsDto(
    bool CanEdit,
    bool CanCancel,
    bool CanMakeOffer,
    bool CanViewOffers,
    bool HasOffered,
    bool CanAcceptOffer,
    bool CanWithdrawOffer,
    bool CanEditOffer);

public sealed record UserSummaryDto(
    Guid Id,
    string FullName,
    string? ProfileImageUrl
   );

public sealed record ItemSpecificationDto(
    string Title,
    string Description,
    string Category,
    string? ImageUrl
    );

public sealed record OfferSummaryDto(
    Guid Id,
    string Description,
    Condition Condition,
    HandoverMethod HandoverMethod,
    OfferStatus OfferStatus,
    TimeSlotDto? AvailableDateDto
    );

public sealed record TimeSlotDto(
    DateTimeOffset StartDate,
    DateTimeOffset EndDate);

File: Client/src/entities/BorrowRequest/BorrowRequestDto.ts
// types.ts

export interface UserSummaryDto {
  id: string;
  fullName: string;
  profileImageUrl?: string;
}

export interface ItemSpecificationDto {
  title: string;
  description: string;
  category: string;
  imageUrl?: string;
}

export interface TimeSlotDto {
  startDate: string;
  endDate: string;
}

export interface OfferSummaryDto {
  id: string;
  description: string;
  condition: string; // Enum string
  handoverMethod: string; // Enum string
  offerStatus: string; // Enum string
  availableDateDto?: TimeSlotDto;
}

export interface BorrowRequestActionsDto {
  canEdit: boolean;
  canCancel: boolean;
  canMakeOffer: boolean;
  canViewOffers: boolean;
  hasOffered: boolean;
  canAcceptOffer: boolean;
  canWithdrawOffer: boolean;
  canEditOffer: boolean;
}

export interface BorrowRequestDto {
  id: string;
  borrower: UserSummaryDto;
  itemNeeded: ItemSpecificationDto;
  neededDates: TimeSlotDto;
  borrowRequestStatus: string; // Enum
  createdAt: string;
  offerCount: number;
  offerSummaryDto?: OfferSummaryDto;
  borrowRequestActionsDto: BorrowRequestActionsDto;
}
